File: app/api-house/app/Models/Room.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable([
    'boarding_house_id',
    'name',
    'status',
    'rent_amount',
    'deposit_amount',
    'electricity_rate',
    'water_rate',
    'wifi_fee',
    'trash_fee',
    'parking_fee',
    'initial_electricity_reading',
    'initial_water_reading',
])]
class Room extends Model
{
    use HasFactory;

    public function boardingHouse(): BelongsTo
    {
        return $this->belongsTo(BoardingHouse::class);
    }

    public function roomTenants(): HasMany
    {
        return $this->hasMany(RoomTenant::class);
    }

    public function meterReadings(): HasMany
    {
        return $this->hasMany(MeterReading::class);
    }

    public function invoices(): HasMany
    {
        return $this->hasMany(Invoice::class);
    }
}

File: app/app-house/lib/models/room.dart
class Room {
  final int id;
  final int boardingHouseId;
  final String name;
  final String status;
  final int rentAmount;
  final int electricityRate;
  final int waterRate;
  final int wifiFee;
  final int trashFee;
  final int parkingFee;

  const Room({
    required this.id,
    required this.boardingHouseId,
    required this.name,
    required this.status,
    required this.rentAmount,
    required this.electricityRate,
    required this.waterRate,
    required this.wifiFee,
    required this.trashFee,
    required this.parkingFee,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: (json['id'] as num).toInt(),
      boardingHouseId: (json['boarding_house_id'] as num).toInt(),
      name: (json['name'] as String?) ?? '',
      status: (json['status'] as String?) ?? 'vacant',
      rentAmount: (json['rent_amount'] as num?)?.toInt() ?? 0,
      electricityRate: (json['electricity_rate'] as num?)?.toInt() ?? 0,
      waterRate: (json['water_rate'] as num?)?.toInt() ?? 0,
      wifiFee: (json['wifi_fee'] as num?)?.toInt() ?? 0,
      trashFee: (json['trash_fee'] as num?)?.toInt() ?? 0,
      parkingFee: (json['parking_fee'] as num?)?.toInt() ?? 0,
    );
  }
}

